src/core/dyad_core.c

```diff
@@ -0,0 +1,622 @@
+#include "dyad_core.h"
+#include "dyad_err.h"
+#include "utils.h"
+#include "murmur3.h"
+
+#include <libgen.h>
+#include <limits.h>
+#include <string.h>
+#include <unistd.h>
+
+const struct dyad_ctx dyad_ctx_default = {
+    NULL,
+    false,
+    false,
+    false,
+    true,
+    false,
+    false,
+    3u,
+    1024u,
+    0u,
+    NULL,
+    NULL
+};
+
+int validate_context(dyad_ctx_t *ctx, bool check_cons, bool check_prod)
+{
+    if (!ctx || !ctx->h)
```

Comment:
Styling-wise, the opening brace better be at the end of the same line with `if` unless the line becomes too long, or the block has too many lines.